test/pkg/integration/integration.go

```diff
@@ -501,3 +503,27 @@ func isPodReady(s *corev1.PodStatus) bool {
 
 	return false
 }
+
+func FindVolumeSnapshot(instanceID, namespace string, client klient.Client) (string, string, error) {
+	vsClient, err := volumesnapshotv1.NewForConfig(client.RESTConfig())
+	if err != nil {
+		return "", "", err
+	}
+
+	volumeSnapshots, err := vsClient.SnapshotV1().VolumeSnapshots(namespace).List(context.Background(), metav1.ListOptions{
```

Comment:
Is this client a sure success? Because pod port forwarding etc. is unstable and will try to retry.

Reply:
It is not using the port-forward function https://github.com/gitpod-io/gitpod/blob/2871bdc1b853dbdef502c54cf950e7319babef61/test/pkg/integration/common/port_forward.go#L31

It's a client that uses kubeconfig to access the Kubernetes cluster.

Reply:
The question was not good. Is this client stable in the face of unstable networks and lack of resources?

Reply:
Ugh, if the network is unstable _or_ the kube-apiserver is out of resources, then yes, the client can't connect to the server.

Reply:
I'm not saying perfect, but what about adding retrying to tolerate such an environment?

Reply:
I see. Currently, all the kubernetes client does not implement the retry within the integration tests https://github.com/gitpod-io/gitpod/blob/main/test/pkg/integration/integration.go, can we handle it in other PR?

Reply:
Make sense